rfcs/2023-rsp-testing-utils.md

```diff
@@ -0,0 +1,113 @@
+
+```jsx
+render(
+  <Provider theme={theme}>
+    <Picker label="Test" data-testid="test" onSelectionChange={onSelectionChange} onOpenChange={onOpenChange}>
+      <Item key="one">One</Item>
+      <Item key="two">Two</Item>
+      <Item key="three">Three</Item>
+    </Picker>
+  </Provider>
+);
+
+let picker = new SelectTester({element: screen.getByTestId('test'), timerType: 'real'});
+await picker.selectOption('Three');
+expect(picker.trigger).toHaveTextContent('Three');
+```
+
+Since these classes are specific to each ARIA pattern, they should work for React Aria Components (RAC) as well as React Spectrum (RSP) components and thus need to handle possible differences in what `element` is provided. For instance, the `data-testid` passed to a RSP Picker is attached to the trigger button element but the `data-testid` for a RAC Select could be the one attached to the wrapper element instead. Alternatively, we could just standardize what `element` is expected to be provided to each of the util classes (aka make the user pass in the Select's trigger button element).
+
+These util classes could live in the `@react-aria/test-util` package and re-exported from `react-aria-components`.
+
+## Documentation
+
+Documentation for the first two parts described above should go in the existing "Testing" page that we have [in the docs](https://react-spectrum.adobe.com/react-spectrum/testing.html#testing) already. We'll want to consider having a similar page in the React Aria Component/React Aria docs sections as well or moving that page up in hierarchy so it lives on the [main page](https://react-spectrum.adobe.com/index.html). The test util classes could also be documented in the same page as the rest of the testing info. I think we should also include a testing section in each of the individual component pages that highlights what test util class can be used (aka mapping Picker to SelectTester), documents the class's methods and options, and links back to the overall testing FAQ for common issues/gotchas that are related to said component. The main testing page should also show a high level example of using a test util class in a example test.
```

Comment:
We could also consider having a testing section in the individual component pages if there are specific things to cover for a given component. For example I imagine testing table might be complex. Once we have the test utils for certain aria patterns it might be nice to show an example of that, or at least link to it from each component as well.